stable/traefik/templates/deployment.yaml

```diff
@@ -94,11 +94,15 @@ spec:
         {{- if or (and .Values.acme.enabled (eq .Values.acme.challengeType "dns-01") .Values.acme.dnsProvider.name) .Values.env }}
         env:
         {{- range $k, $v := (index .Values.acme.dnsProvider .Values.acme.dnsProvider.name) }}
-          {{- if $v }}
+          {{- if or $v $.Values.acme.dnsProvider.existingSecretName }}
```

Comment:
Is this `or` needed? If we are iterating through the loop, shouldn't this always be true? leading to multiple mounts with the same keyref?

Reply:
I think the `or` is needed and multiple mounts with the same keyref should not happen. I'm not very confident with templating so let me explain the logic.

#### Previous scenario (`{{- if $v }}`) 
If `dnsProvider.name` is set:
* We retrieve a dictionary of environment variable names and their values for the specified dnsProvider
* We iterate over those environment variables. For each environment variable:
  * If the value is set we create an `env` item with the environment variable name as key. This environment variable is retrieved from the secret `{{ template "traefik.fullname" $ }}-dnsprovider-config`, under a key with the same name as the environment variable
  * If the value is not set we skip it

#### PR scenario  ( `{{- if or $v $.Values.acme.dnsProvider.existingSecretName }}`)
If `dnsProvider.name` is set:
* We retrieve a dictionary of environment variable names and their values for the specified dnsProvider
* We iterate over those environment variables. For each environment variable:
  * And `env` item is created if either
    * value is set for that environment variable (previous behaviour)
    * `existingSecretName` is specified (that would mean we want to use an existingSecretName as source for those environment values)
  * If an `env` item is created for the environment variable, its value will be:
    * The value contained in the corresponding `existingSecret` (if `existingSecret` is set)
    * The value specified in values file otherwise

There would be an error of course if an `existingSecret` is specified, and that `existingSecret` doesn't contain the required keys but I would say that is the intended behaviour as those variables seem to be mandatory for using the provider.

Reply:
@dtomcej would you require additional clarification or should I mark the conversation as resolved?